crypto/rand/rand_lib.c

```diff
@@ -149,14 +149,13 @@ int RAND_poll(void)
         ret = 1;
      err:
         ossl_rand_pool_free(pool);
+        return ret;
     }
-    return ret;
-# else
+# endif
     static const char salt[] = "polling";
```

Comment:
You'll have to move this line up before the `# ifndef OPENSSL_NO_DEPRECATED_3_0` otherwise you would break ansi build.

Reply:
There's an -ansi -pedantic build?!  Fine.  (I thought this might happen.)